### DISCUSSION TITLE
What is the public api for sqlalchemy?

### DISCUSSION THREAD

**Discussion by pinkfrog9** [sqlalchemy/sqlalchemy#8893] [Usage Questions] [answered]
For example, for `Column`, the canonical path is `sqlalchemy.schema.Column`.
However, one normally uses `sqlalchemy.Column`.

The issue is that, in the documentation on `Column`, it is: 
<img width="825" alt="image" src="https://user-images.githubusercontent.com/669045/204429488-9884ec32-090c-4751-ba1f-ea6f6b452d69.png">

How can one know which one is the correct path to use? *Especially considering less used ones instead of Column*?

**Comment by zzzeek** [comment, marked as answer]
the sphinx documentation is a work in progress and it is not always easy to get it to be consistent about module naming.    Additionally, tools like Pylance tend to pop things up from their "real" module, like "sqlalchemy.schema.Column" above.

for SQLAlchemy 2.0, the goal is that *all* objects in the public API will be from "sqlalchemy.", "sqlalchemy.orm." "sqlalchemy.ext" the sub-modules of "sqlalchemy.ext" one level deep, "sqlalchemy.dialects" and the submodules of "sqlalchemy.dialects" one level deep.   This is mostly how it is in 1.4 however there is a lot of historical use of "sqlalchemy.sql.expression", etc.

Overall the best guide to where imports should come from should be based on the examples given in the documentation for various API features.  the auto-generated Sphinx documentation is not that accurate as of yet.

**Reply by pinkfrog9** [reply]
What about 'sqlalchemy.sql', is it deemed public?

**Reply by CaselIT** [reply]
`sqlalchemy.sql` should be more or less equivalent to `sqlalchemy` since most (if not all) exported object are re-exported by the init.

**Reply by CaselIT** [reply]
I've added #8894 to track this
